### PR TITLE
update parent pom and necessary modifications for compatibility

### PR DESCRIPTION
**Pull request by BobDu**
Update parent pom and make the necessary modifications for compatibility.
And manual run test in unity 2022 lts version.

### Testing done

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by BobDu** [comment]
@lacostej 
Sorry to bother you

I see you not continuing use this plugin in https://forum.unity.com/threads/automating-builds-with-jenkins.319169/

Would you be interested in continuing to maintain this plugin?

If needed, I am willing to join the team of unity3d plugin maintainers to continue maintaining this plugin.

**Comment by BobDu** [comment]
ping @lsegal 

I would like to take over the maintenance of this plugin following this process
https://www.jenkins.io/doc/developer/plugin-governance/adopt-a-plugin/

Hope to get your feedback. Thanks.

**Comment by basil** [comment]
Would also be nice to [incrementalify](https://www.jenkins.io/doc/developer/plugin-development/incrementals/) the plugin and enable Dependabot.

File: pom.xml [hunk 1]
```diff
@@ -1,102 +1,94 @@
...
+          </pom>
+        </configuration>
+        <executions>
+          <execution>
+            <goals>
+              <goal>check</goal>
+            </goals>
+            <phase>compile</phase>
+          </execution>
+        </executions>
```
Comment (basil) [inline pom.xml:90]: Unnecessary duplicate code.

File: pom.xml [hunk 2]
```diff
@@ -1,102 +1,94 @@
...
+    <plugins>
+      <plugin>
+        <groupId>com.diffplug.spotless</groupId>
+        <artifactId>spotless-maven-plugin</artifactId>
+        <configuration>
+          <java>
+            <includes>
+              <include>dummy</include>
+            </includes>
+          </java>
```
Comment (basil) [inline pom.xml:73]: Cleaner not to refer to a dummy include:

```
          <java combine.self="override"/>
```

**Comment by BobDu** [comment]
@basil 
thanks for your guidance
I will further improve it in subsequent PRs

There are also some things that are bothering me and I would like your advice.
I hope to enable CI system for this plugin.

But currently, when running unit tests, you need to have a unity environment on your machine.
So making this plugin more difficult to enable CI systems. Do you have any suggestions or know of other plugins that have similar problems and how they solved them?

**Comment by basil** [comment]
I think (?) if you run the build on a full VM (i.e., `buildPlugin(useContainerAgent: false)` that you can install any packages you want with `sudo apt install […]`. If not then you should be able to file a ticket at https://github.com/jenkins-infra/helpdesk for guidance with our container / VM images.

**Pull request merged**